### ISSUE TITLE
stop streaming without closing camera

### ISSUE THREAD

**Issue by tareqalbeesh** [pedroSG94/RootEncoder#848]
I am trying to stop both the streaming and the recording without closing the camera, 
and that's because I have an ImageReader's Surface that is giving me frames all the time, 
I managed to do so by not closing camera when I call `stopStream()`,
it worked, 
but after couple of seconds I am getting the error: 

`E/BufferQueueProducer: [SurfaceTexture-1-2785-2] queueBuffer: BufferQueue has been abandoned`

**Comment by pedroSG94** [comment]
Did you tried recreate ImageReader and set this new one?

**Comment by tareqalbeesh** [comment]
do you mean recreating a new ImageReader after closing the camera recording and streaming ? 
if so, how to do this task in the background without any preview using your library ?

**Comment by pedroSG94** [comment]
You can close camera and create that imagereader before reopen camera

**Comment by tareqalbeesh** [comment]
I did exactly that and it's working perfectly,
is possible to do the same logic without even closing the camera at all? 
so the imageReader is always reading the frames even thought i am not recording or streaming (a possible solution I did is to keep the encoder running all the time, but it was cpu heavy for nothing, and a bit buggy)

**Comment by pedroSG94** [comment]
Maybe something similar to this:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/encoder/src/main/java/com/pedro/encoder/input/video/Camera2ApiManager.java#L742
You cnac eate a new method where you call stopRepeating, change surface list (including all one and your imagereader) and setRepeatingRequest again to re start.

**Comment by tareqalbeesh** [comment]
I tried both this method and the stopping the camera and reopening it, and the latency was almost the same, is this behavior expected?
to be more specific,
I tried first,
when I stop the streaming and recording, the encoder would become useless, so i would close the camera and reopen it with just ImageReader Surface, and it took about 1.5 seconds to do this process.

the second thing I tried was your latest suggestion,
When I stop the streaming and recording , I would stop repeating the `CameraCaptureSession`, then I would change just the surfaces (keep just the ImageReader's surface), and the time to do this task was almost identical to the previous experience.
